src/pages/communication-page/__tests__/communication.spec.ts

```diff
@@ -117,14 +113,37 @@ describe('CommunicationPage', () => {
     });
 
     describe('DOM', () => {
-      it('should display the correct address fields', () => {
+      const address = {
+        candidateAddress: {
+          addressLine1: 'Somewhere',
+          addressLine2: 'Someplace',
+          addressLine3: 'Sometown',
+          addressLine4: 'Somecity',
+          addressLine5: 'Somecountry',
+          postcode: 'AB12 3CD',
+        },
+      };
+
+      it('should display the no address fields when address is empty', () => {
         expect(fixture.debugElement.query(By.css('#addressLine1'))).toBeNull();
-        expect(fixture.debugElement.query(By.css('#addressLine2'))).toBeDefined();
-        expect(fixture.debugElement.query(By.css('#addressLine3'))).toBeDefined();
+        expect(fixture.debugElement.query(By.css('#addressLine2'))).toBeNull();
+        expect(fixture.debugElement.query(By.css('#addressLine3'))).toBeNull();
         expect(fixture.debugElement.query(By.css('#addressLine4'))).toBeNull();
         expect(fixture.debugElement.query(By.css('#addressLine5'))).toBeNull();
+        expect(fixture.debugElement.query(By.css('#postcode'))).toBeNull();
+      });
+
+      mockCandidate.candidateAddress = address;
+
+      it('should display the correct address fields when address populated', () => {
+        expect(fixture.debugElement.query(By.css('#addressLine1'))).toBeDefined();
```

Comment:
Checking for the result of fixture query `toBeDefined` is a misleading test, as failure to find the element will yield `null`.